inst/templates/tidy_contributing.md

```diff
@@ -0,0 +1,147 @@
+[styler](https://CRAN.R-project.org/package=styler) package to apply the style 
+guide automatically.
+
+
+Be careful to only make style changes to the code you are contributing. If you
+find that there is a lot of code that doesn't meet the style guide, it would be
+better to file an issue or a separate PR to fix that first.
+
+### Document
+
+We use [roxygen2](https://cran.r-project.org/package=roxygen2),
+specifically with the [Markdown syntax](https://cran.r-project.org/web/packages/roxygen2/vignettes/markdown.html),
+to create `NAMESPACE` and all `.Rd` files. All edits to documentation
+should be done in roxygen comments above the associated function or
+object.
+
+``` r
+devtools::document()
+```
+
+See the `RoxygenNote` in [DESCRIPTION](DESCRIPTION) for the version of
+roxygen2 being used. 
+
+### Test
+
+We use [testthat](https://cran.r-project.org/package=testthat). Contributions
+with test cases are easier to accept. If you are not sure what parts of your
+code is covered by tests, run the following to get a local coverage report of
+the package so you can see exactly what lines are not covered in the project.
+
+``` r
+devtools::test_coverage()
+```
+
+### NEWS
+
+For user-facing changes, add a bullet to `NEWS.md` that concisely describes
+the change. Small tweaks to the documentation do not need a bullet. The format
+should include your GitHub username, and links to relevant issue(s)/PR(s), as
+seen below.
+
+```md
+* `function_name()` followed by brief description of change (#issue-num, @your-github-user-name)
+```
+
+### Re-check
+
+Before submitting your changes, make sure that the package either still
+passes `R CMD check`, or that the warnings and/or notes have not _changed_
+as a result of your edits.
+
+### Commit
+
+When you've made your changes, write a clear commit message describing what
+you've done. If you've fixed or closed an issue, make sure to include keywords
+(e.g. `fixes #101`) at the end of your commit message (not in its
+title) to automatically close the issue when the PR is merged.
+
+## Push and pull
+
+Once you've pushed your commit(s) to _your_ fork, you're ready to make the pull
+request. Pull requests should have descriptive titles to remind 
+reviewers/maintainers what the PR is about. You can easily view what exact 
+changes you are proposing using either the Git diff view in RStudio, or the 
+[branch comparison view](https://help.github.com/articles/creating-a-pull-request/) 
+you'll be taken to when you go to create a new PR. If the PR is related to an 
+issue, provide the issue number and slug in the _description_ using auto-linking 
+syntax (e.g. `#15`).
+
+### Review, revise, repeat
+
+Since tidyverse development happens in waves, the latency period between
+submitting your PR and its reviewed may vary. When a maintainer does review
+your contribution, be sure to use the same conventions described here with any
+revision commits.
+
+## Resources
+
```

Comment:
I'd add a link to @hadley's R Packages book. Probably with specific pointers to chapters like [checking](http://r-pkgs.had.co.nz/check.html), [documentation](http://r-pkgs.had.co.nz/man.html), and [tests](http://r-pkgs.had.co.nz/tests.html).

Reply:
These are great, Jenny. Thanks for your comments. I added these, as well as a few other GitHub links along the way. 

I believe (@hadley, correct me if I'm wrong) the target audience for this should be somewhat familiar with Git/GitHub, and that this will complement the expanded guide for beginners (still in production 😬). The latter guide is decidedly grounded in RStudio (or at least the screenshots are), and I tried to keep this one somewhat IDE-agnostic (without featuring anything that would scare a relative novice off).

Reply:
That's right